packages/expo-router/src/layouts/StackClient.tsx

```diff
@@ -27,8 +29,45 @@ type GetId = NonNullable<RouterConfigOptions['routeGetIdList'][string]>;
 
 const NativeStackNavigator = createNativeStackNavigator().Navigator;
 
+/**
+ * We extend NativeStackNavigationOptions with our custom props
+ * to allow for several extra props to be used on web, like modalWidth
+ */
+export type ExtendedStackNavigationOptions = NativeStackNavigationOptions & {
+  /**
+   * Override the width of the modal (px or percentage). Only applies on web platform.
+   * @platform web
+   */
+  modalWidth?: number | string;
+  /**
+   * Override the height of the modal (px or percentage). Applies on web desktop.
```

Comment:
Should document that these can be CSS variables on web to support responsive design.

Reply:
When you say document, do you mean in code or the Expo Router docs?